_posts/2024-07-19-Using-Redis-with-Sidecar.md

```diff
@@ -0,0 +1,179 @@
+
+1. **Containerize Your .NET Application**
+    Create a Dockerfile in the root directory of your project. The Dockerfile for this project is [here](https://github.com/Azure-Samples/sidecar-samples/blob/main/Redis.Core.WebApi-master/Redis.Core.WebApi/Dockerfile).
+2. **Build and Push the Docker Image**
+    Build the image and push it to your preferred container registry, such as Azure Container Registry, Docker Hub, or a private registry.
+
+    ```bash
+    docker build -t <your-registry>/<your-image-name>:<tag> .
+    docker push <your-registry>/<your-image-name>:<tag>
+    ```
+
+    We would also recommend that you push the local redis image to ACR. This would allow you to take advantage of securing your resources using Managed Identity. Please refer to this [documentation](https://learn.microsoft.com/en-us/azure/container-registry/container-registry-get-started-docker-cli?tabs=azure-cli).
+3. **Create a New Linux Web App in Azure**
+    Create a new Linux Web App from the portal and choose the options for Container and Linux.
+    ![Create web app]({{site.baseurl}}/media/2024/07/CreateWebApp.jpg)
+
+    On the Container tab, make sure that Sidecar support is Enabled.
+
+    Specify the details of your application image.
+    ![Create web app]({{site.baseurl}}/media/2024/07/AddContainer.jpg)
+
+    *Note: We strongly recommend enabling [Managed Identity](https://learn.microsoft.com/en-us/azure/app-service/overview-managed-identity?tabs=portal%2Chttp) for your Azure resources.*
+
+4. **Add Redis Sidecar**
+    Go to the Deployment Center for your application and add a sidecar container.
+    ![Add Redis sidecar]({{site.baseurl}}/media/2024/07/add-redis-container.jpg)
+
+    This [document](https://learn.microsoft.com/en-us/azure/app-service/tutorial-custom-container-sidecar) tells you how to add sidecars, step-by-step.
+
+5. **Verify the deployment**
+    Once your deployment is complete, you can browse to `<your application url>/weatherforecast`
+
+    You should see output like this
+    ![Page output]({{site.baseurl}}/media/2024/07/website-output.jpg)
+
+    Refresh the page a few times. You can go to the `Log Stream` in the Azure Portal to see logs indicating whether the response is coming from the database or the cache.
+    ![Log stream]({{site.baseurl}}/media/2024/07/redis-logs.jpg)
+
+## Caveat for Multi-Instance Workloads
+
+There is a caveat for this application. We are using a local instance of Redis. If you scale out your application to multiple instances, you will notice that the response does not always come from the cache, depending on which instance your request is being routed to. For multi-instance workloads, we recommend using [Azure Cache for Redis](https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-overview), which provides a distributed, scalable, and high-performance caching solution.
```

Comment:
remove */en-us/* from links this help route folks to localized content

Reply:
Done